source/helpers/CaseDataConverter.js

```diff
@@ -91,16 +91,12 @@ export const convertAnswersToArray = (data, formQuestions) => {
           Object.entries(childItems).forEach((childItem) => {
             const [repeaterItemId, repeaterItemValue] = childItem;
             const repeaterFieldItem = other.inputs.find((obj) => obj.id === repeaterItemId);
-            const { tags } = repeaterFieldItem;
+            const { tags: tagsInRepeaterField } = repeaterFieldItem;
             let newTags = [];
             if (Array.isArray(tags)) {
-              const dynamicTagRegex = new RegExp('.+:x');
-              newTags = tags.map((tag) => {
-                if (dynamicTagRegex.test(tag)) {
-                  const strArray = tag.split(':');
-                  return `${strArray?.[0] || tag}:${childFieldId}`;
-                }
-                return tag;
+              newTags = tagsInRepeaterField.map((tag) => {
+                  const updatedTag = tag.replace(':x', `:${childFieldId}`)
```

Comment:
This will replace even if the tag part starts with x (like `tag:xyz`), however replace can take a regex so maybe something like `:x(?=(:.*)?$)` (assuming positive lookaheads are supported - not sure)?